common/base-api.js

```diff
@@ -294,17 +314,21 @@ class BaseApi {
     return new Promise((resolve) => {
       if (this.persistent) { // local storage
         if (key in localStorage) {
-          const obj = JSON.parse(localStorage.getItem(key));
-          if (this._timeout === null && obj.kt === null) {
-            resolve(true);
-          } else {
-            if (this._isCacheStale(obj)) {
-              this.removeKey(key).then(() => {
-                resolve(false);
-              });
-            } else {
+          try {
+            const obj = JSON.parse(localStorage.getItem(key));
+            if (this._timeout === null && obj.kt === null) {
               resolve(true);
+            } else {
+              if (this._isCacheStale(obj)) {
+                this.removeKey(key).then(() => {
+                  resolve(false);
+                });
+              } else {
+                resolve(true);
+              }
             }
+          } catch (e) {
+            resolve(false);
```

Comment:
Same logic as the keyExistsSync() method.

We're in the "key in localStorage" block. It's not valid JSON, but nonetheless the key does exist.

Please update to "resolve(true)" in this catch block.